### PR TITLE
Change separator from \1 to \u0001 

### PR DESCRIPTION
**Pull request by oeddyo**
To remove warning info for scala 2.11

### CODE REVIEW

**Comment by oeddyo** [comment]
Seems there're some format issues. I was using 

./sbt ++2.11.5 clean test

How shall I automatically get a nicely formatted code?

File: scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala [hunk 1]
```diff
@@ -39,9 +39,9 @@ object CumulativeSum {
     def cumulativeSum(
       implicit sg: Semigroup[V], ordU: Ordering[U], ordK: Ordering[K]): SortedGrouped[K, (U, V)] = {
       pipe.group
-        .sortBy { case (u: U, _) => u }
+        .sortBy { case (u, _) => u }
```
Comment (isnotinvain) [inline scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala]: why drop the types?

Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala]: Those give warnings (I think) by the compiler since scala can't see that the type MUST be U, and U is also erased. Scala type inference does not work great with case matches.

Comment (jcoveney) [inline scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala]: scala can't verify the type, and I imagine it throws an error

File: scalding-core/src/main/scala/com/twitter/scalding/Config.scala [hunk 2]
```diff
@@ -267,8 +267,8 @@ trait Config {
    */
   def addReducerEstimator(clsName: String): Config =
     update(Config.ReducerEstimators) {
-      case None => Some(clsName) -> ()
-      case Some(lst) => Some(clsName + "," + lst) -> ()
+      case None => Some(clsName) -> Unit
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/Config.scala]: This does not seem like an error to me. `Unit` is the type or the companion object where as `()` is the value of type `Unit`. This code is different, although it does not matter because we discard the side value at the end. 

Why the change?

**Comment by johnynek** [comment]
Looks good to me other than the `()` to `Unit` change. I want to know more about that.

File: scalding-core/src/main/scala/com/twitter/scalding/Config.scala (hunk 2 above)
Comment (oeddyo) [inline scalding-core/src/main/scala/com/twitter/scalding/Config.scala]: @johnynek I think I'm mixing two PR. One is to remove as many warnings as possible(https://github.com/twitter/scalding/issues/1139). For that purpose I'm seeing scala 2.11 gives warning when using () instead of Unit

[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/Config.scala:230: Adaptation of argument list by inserting () has been deprecated: this is unlikely to be what you want.
[warn]         signature: ArrowAssoc.->[B](y: B): (A, B)
[warn]   given arguments: <none>
[warn]  after adaptation: ArrowAssoc.->((): Unit)
[warn]       case Some(lst) => Some(clsName + "," + lst) -> ()


I googled around and found this:
http://stackoverflow.com/questions/25351988/adaptation-of-argument-list-by-inserting-has-been-deprecated

File: scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala (hunk 1 above)
Comment (oeddyo) [inline scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala]: @isnotinvain Dropping the types since there're warnings about type erasure here. 



[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala:42: abstract type pattern U is unchecked since it is eliminated by erasure
[warn]         .sortBy { case (u: U, _) => u }
[warn]                            ^
[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala:44: abstract type pattern U is unchecked since it is eliminated by erasure
[warn]           case (acc, (u: U, v: V)) =>
[warn]                          ^
[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala:44: abstract type pattern V is unchecked since it is eliminated by erasure
[warn]           case (acc, (u: U, v: V)) =>
[warn]                                ^
[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala:66: abstract type pattern U is unchecked since it is eliminated by erasure
[warn]         .map { case (k, (u: U, v: V)) => (k, partition(u)) -> v }
[warn]                             ^
[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala:66: abstract type pattern V is unchecked since it is eliminated by erasure
[warn]         .map { case (k, (u: U, v: V)) => (k, partition(u)) -> v }
[warn]                                   ^
[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala:90: abstract type pattern U is unchecked since it is eliminated by erasure
[warn]           case (k, (u: U, v: V)) =>
[warn]                        ^
[warn] /Users/exie/learn/ScaldingGithub/scalding/scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala:90: abstract type pattern V is unchecked since it is eliminated by erasure
[warn]           case (k, (u: U, v: V)) =>

File: scalding-core/src/main/scala/com/twitter/scalding/Config.scala (hunk 2 above)
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/Config.scala]: Ahh, this is what we want here. So, let's change it to:

```scala
(Some(s"$clsName,$lst"), ())
(Some(clsName), ())
```
and just not use the `->` enrichment.

File: scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala (hunk 1 above)
Comment (isnotinvain) [inline scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala]: can it tell enough to know the "shape" is `(k, (u, v))`?

This is one thing I'm often confused about with case matching in scala, how to make sure you aren't accidentally doing reflection in a case match in a way that will never even match.

**Comment by isnotinvain** [comment]
LGTM other than the Unit change, it should remain `()`
Any idea why some of the tests that weren't changed go picked up in the diff?

**Comment by oeddyo** [comment]
@isnotinvain Sure. I'll chnage it to (). Not really sure why the test missed this

**Comment by isnotinvain** [comment]
@oeddyo well because in this case, any value at all would work but `Unit` makes a little less sense than `()` -- it's sort of like using `java.lang.Void.class` instead of `null`

**Comment by oeddyo** [comment]
@isnotinvain Thanks for the explanation :)

**Comment by oeddyo** [comment]
@johnynek @isnotinvain Fixed. Shall I merge?

**Comment by johnynek** [comment]
Thanks, @oeddyo!

**Comment by johnynek** [comment]
closes #1215
